src/extension.ts

```diff
@@ -164,23 +171,29 @@ export function activate(context: vscode.ExtensionContext): void {
         if (!node) {
             try {
                 node = <IAzureNode<WebAppTreeItem>>await tree.showNodePicker(WebAppTreeItem.contextValue);
+
             } catch (err2) {
                 if (parseError(err2).isUserCancelledError) {
                     this.properties.cancelStep = `showNodePicker:${WebAppTreeItem.contextValue}`;
                 }
                 throw err2;
             }
         }
-
         try {
-            await node.treeItem.deploy(fsPath, outputChannel, ui, reporter, extensionPrefix, true, this.properties);
+            if (newApp && newApp.id === node.id) {
+                // if the node selected for deployment is the same as the one that was created from the picker, stifle the confirmDeployment dialog
+                confirmDeployment = false;
+            }
+            await node.treeItem.deploy(fsPath, outputChannel, ui, reporter, extensionPrefix, confirmDeployment, this.properties);
         } catch (err) {
             if (parseError(err).isUserCancelledError) {
                 throw err;
             }
             const appServicePlan = await node.treeItem.client.getAppServicePlan();
             this.properties.servicePlan = appServicePlan.sku.size;
             throw err;
+        } finally {
```

Comment:
I liked that you used try/finally, but it only works if onNodeCreatedFromQuickPickDisposable is created right before the 'try' starts. For example, if the user cancels in showNodePicker, this would never get disposed in the current implementation

Reply:
Does that mean I should call dispose twice, once in the finally and once in the catch here
`catch (err2) {
                if (parseError(err2).isUserCancelledError) {
                    this.properties.cancelStep = `showNodePicker:${WebAppTreeItem.contextValue}`;
                }
                throw err2;
            }`

Reply:
No that means you need another try/finally that wraps _all_ of the code between `const onNodeCreatedFromQuickPickDisposable: vscode.Disposable = ...` and when you call dispose

Reply:
Yep you got it